packages/core/components/icon/index.js

```diff
@@ -1,50 +1,63 @@
-import React from "react";
+import React, { useContext } from "react";
 import PropTypes from "prop-types";
-import styled, { withTheme } from "styled-components";
-import { size } from "styled-system";
-import Text from "../text";
+import styled, { ThemeContext } from "styled-components";
+import { themeGet } from "styled-system";
+import { COMMON, FLEX_ITEM, LAYOUT, MISC, POSITION } from "../../constants";
 
-/* helpful links
-  SVG optimization
-  - https://github.com/svg/svgo
-  SVG Transformation or inspection
-  - https://github.com/elrumordelaluz/svgson-next
-  - https://github.com/Angelmmiguel/svgi
-*/
-const StyledIcon = styled(Text)`
-  ${size}
+const StyledIcon = styled.svg`
   vertical-align: middle;
+  font-size: inherit;
+  line-height: inherit;
+
+  ${p =>
+    !p.size &&
+    css`
+      width: 1em;
+      height: 1em;
+    `}
+
+  ${themeGet("icon.styles")};
+  ${COMMON}
+  ${FLEX_ITEM}
+  ${LAYOUT}
+  ${MISC}
+  ${POSITION}
 `;
 
-const Icon = ({ children, ...props }) => {
-  const icon = props.theme.icons[props.name];
+const Icon = ({ ...p }) => {
```

Comment:
Recommend changing to `({path, ...p})` to extract `path` so it doesn't get passed as an attribute to the `<svg>` on line 37.